### ISSUE TITLE
About setting of WSDDN

### ISSUE THREAD

**Issue by suilin0432** [NVlabs/wetectron#47]
we followed the setting in detectron2.

_Originally posted by @jason718 in https://github.com/NVlabs/wetectron/issues/46#issuecomment-812829986_

Fine...
This is really a good job that the Mist algorithm works well when I transfer the design to my codebase with slightly modification.
But it's really strange that the performance could drop such a lot when using gpus less than 8 with the same batch_size settings and the performance is influnced by the random seed seriously in my experiments. I compare this codebase with others and I do not find any serious problem, so I guess maybe influnced by the large base_lr or the initialization strategy. I believe wetectron will be a popular framework for WSOD if you could refine details to make it more stable~

By the way, could you please offer a setting about the vanilla WSDDN, i have tried several times and the performance is not good.
Thanks a lot~

**Comment by jason718** [comment]
Thanks for your good word. We received lots of feedback about the small batch-size thing. I also not sure what's the exact reason. Randomness is indeed a headache, maybe that's the thing?

in terms of vanilla WSDDN, are you asking about the learning rate, batch size, etc?

**Comment by suilin0432** [comment]
Thanks for your reply~
Yeah, about learning rate, batch size and steps. It's better to show me the config file you have ever tried. 
But, I still wonder it's strange to use 21 classes in MIL branch instead of using 20 classes in MIL branch and 21 classes in oicr branch.

**Comment by jason718** [comment]
I don't think we reported any WSDDN number in our paper. But I found a very old config file that I might have used at some time point

```
MODEL:
  META_ARCHITECTURE: "GeneralizedRCNN"
  WEIGHT: "catalog://VGGImageNetPretrained/JCJOHNS/VGG-16"
  FASTER_RCNN: False
  BACKBONE:
    CONV_BODY: "VGG16"
  WSOD_ON: True
  ROI_HEADS:
    SCORE_THRESH: 0.0
    NMS: 0.5
  ROI_BOX_HEAD:
    NUM_CLASSES: 21
    POOLER_METHOD: "ROIPool"
    POOLER_RESOLUTION: 7
    POOLER_SCALES: (0.0625,)
    FEATURE_EXTRACTOR: "VGG16.roi_head"
  ROI_WEAK_HEAD:
    PREDICTOR: "WSDDNPredictor"
    LOSS: "WSDDNLoss"
DATASETS:
  TRAIN: ("voc_2007_train", "voc_2007_val")
  TEST: ("voc_2007_test",)
PROPOSAL_FILES:
  TRAIN: ('proposal/SS-voc_2007_train-boxes.pkl','proposal/SS-voc_2007_val-boxes.pkl')
  TEST:  ('proposal/SS-voc_2007_test-boxes.pkl',)
INPUT:
  MIN_SIZE_TRAIN: (480, 576, 688, 864, 1000, 1200) 
  MAX_SIZE_TRAIN: 2000
  MIN_SIZE_TEST: 800
  MAX_SIZE_TEST: 2000
SOLVER:
  IMS_PER_BATCH: 8
  BASE_LR: 0.01
  WEIGHT_DECAY: 0.0001
  STEPS: (12500, 17500)
  MAX_ITER: 20000
  CHECKPOINT_PERIOD: 5000
TEST:
  BBOX_AUG:
    HEUR: "AVG"
    ENABLED: True
    H_FLIP: True
    SCALES: (480, 576, 688, 864, 1000, 1200) 
    MAX_SIZE: 2000
    SCALE_H_FLIP: True
```

For the number of classes setting, we didn't try what you suggested. Maybe it's worthy trying?

**Comment by suilin0432** [comment]
Thanks a lot. I think I may try it~
By the way, I have to say that the implementation details of different code repositories in WSOD field are so different.

**Comment by suilin0432** [comment]
Thanks a lot for your reply.
